### PR TITLE
XR: Initialize GPUPicker once scene is ready

### PR DESCRIPTION
**Pull request by audionerd**

### CODE REVIEW

File: src/js/xr/src/SceneManagerXR.js [hunk 1]
```diff
@@ -133,6 +144,19 @@ const SceneContent = connect(
...
     }, [world.backgroundColor])
 
+    const getGpuPicker = () => {
+      if (gpuPicker.current === null) {
+        gpuPicker.current = new GPUPicker(gl)
+      }
+      return gpuPicker.current
+    }
+
+    useEffect(() => {
```
Comment (audionerd) [inline src/js/xr/src/SceneManagerXR.js]: I don't think GPUPicker init needs to move to SceneManagerXR/SceneContent. This `useEffect` could instead be moved into the interactions manager (which can get access to `scene` and `camera`).

File: src/js/xr/src/use-interactions-manager.js [hunk 2]
```diff
@@ -259,17 +258,35 @@ const useInteractionsManager = ({
...
   const gpuPicker = useRef(null)
-
   const getGpuPicker = () => {
     if (gpuPicker.current === null) {
       gpuPicker.current = new GPUPicker(gl)
     }
     return gpuPicker.current
   }
 
+  useEffect(() => {
```
Comment (audionerd) [inline src/js/xr/src/use-interactions-manager.js]: Can you explain this a bit, maybe comment it? We need to create a fake object in the scene to initialize the picker?

Comment (VladStepura) [reply src/js/xr/src/use-interactions-manager.js]: Basically, we need something to pick to initialize shader_log and stuff like that. If there's nothing in pick area(pick pixel) it won't initialize it. So I'm adding an object in front of the camera and picking it, that way there are definitely will be a pick and GPU picker will initialize properly.

Comment (audionerd) [reply src/js/xr/src/use-interactions-manager.js]: Could we move that code into a method of GPUPicker?

Comment (VladStepura) [reply src/js/xr/src/use-interactions-manager.js]: I don't think GPU Picker needs to know about that stuff, it's kind of hack. I can move it to GPU picker but I think the cleaner way to do that is to keep it outside of it and use it as util or something

**Pull request merged**